### ISSUE TITLE
[openbmc] bmcdiscover does not take password defined in password table 

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#3036]
bmcdiscover function was added into xCAT tracked in this issue.   https://github.com/xcat2/xcat-core/issues/2809 

Running the command against a openbmc node using the following build: 
```
[root@stratton01 consoles]# lsxcatd -a
Version 2.13.4 (git commit d2fd44fbdf64834803d4b77a8a1ed0df10e5729d, built Thu May 11 06:15:46 EDT 2017)
This is a Management Node
dbengine=SQLite
```

The password is not taken from the password table, like it is on OpenPower (IPMI managed) 
```
[root@stratton01 consoles]# bmcdiscover --range 50.5.39.1
50.5.39.1,8335-GTA,2100EEA,,,mp,bmc
```

When not providing a password, it fails: 

```
[root@stratton01 consoles]# bmcdiscover --range 10.6.7.254
Warning: Invalid username or password for 10.6.7.254
```
But manually passing in the user/pass, it works...
```
[root@stratton01 consoles]# bmcdiscover --range 10.6.7.254 -u root -p 0penBmc
10.6.7.254,,Y130UF72700T,root,0penBmc,mp,bmc
```

Since we are going against the bmc using restful, we should be able to get the openbmc password from the passwd table.

**Comment by xuweibj** [comment]
Hi @whowutwut 
For bmcdiscover, if there is no username or password input,  will get username and password for ipmi. So in this case, it got wrong username and password. I will try to fix this.

**Comment by xuweibj** [comment]
The issue was fixed in pull request #3037

**Comment by whowutwut** [comment]
Looks good, verified on today's build
```
[root@stratton01 ~]# lsxcatd -v
Version 2.13.5 (git commit 8ea4e5ae19806c890dd9149afa82fd7fd1b9a7fc, built Tue May 23 06:15:36 EDT 2017)
```

Testing against OpenPower (IPMI) 
```
[root@stratton01 ~]#  bmcdiscover --range 50.5.39.1
50.5.39.1,8335-GTA,2100EEA,,,mp,bmc
```
Testing against OpenPower OpenBMC
```
[root@stratton01 ~]# bmcdiscover --range 10.6.7.254
10.6.7.254,,Y130UF72700T,,,mp,bmc
```

Passwords are defined in `passwd` table
